src/v/s3/client.cc

```diff
@@ -443,6 +498,49 @@ ss::future<http::client::response_stream_ref> client::get_object(
       });
 }
 
+ss::future<client::head_object_result> client::head_object(
+  bucket_name const& name,
+  object_key const& key,
+  const ss::lowres_clock::duration& timeout) {
+    auto header = _requestor.make_head_object_request(name, key);
+    if (!header) {
+        return ss::make_exception_future<client::head_object_result>(
+          std::system_error(header.error()));
```

Comment:
is `std::system_error` mostly associated with os-level errors? is that relevant here?